state/payloads_test.go

```diff
@@ -108,6 +92,20 @@ func (s *PayloadsSuite) TestTrackInsertPayload(c *gc.C) {
 	fix.CheckOnePayload(c, desired)
 }
 
+func (s *PayloadsSuite) TestValidationOnTrack(c *gc.C) {
```

Comment:
I would like to see a test that checks we handle a concurrent upgrade & Track, something like:

- unit starts out with payload
- user calls Track
- unit is concurrently upgraded to new charm without payload
- Track should fail because the charm field of the unit changed, causing it to retry the transaction and revalidate